packages/manager/src/utilities/sort-by.test.ts

```diff
@@ -0,0 +1,218 @@
+import {
+  sortByString,
+  sortByUTFDate,
+  sortByNumber,
+  sortByArrayLength,
+  sortByRegionLabel,
+  toComparator,
+} from './sort-by';
+
+describe('sortByString', () => {
+  it('returns -1 when the first value comes first alphabetically', () => {
+    expect(sortByString('hello', 'HELLO')).toBe(-1);
+    expect(sortByString('hello', 'world')).toBe(-1);
+    expect(sortByString('hello', 'world', 'asc')).toBe(-1);
+    expect(sortByString('Hello', 'Hello!')).toBe(-1);
+    expect(sortByString('Linode 1', 'Linode 2')).toBe(-1);
+    expect(sortByString('Linode 1', 'Linode 10')).toBe(-1);
+  });
+
+  it('returns 1 when the second value comes first alphabetically', () => {
+    expect(sortByString('HELLO', 'hello')).toBe(1);
+    expect(sortByString('world', 'hello')).toBe(1);
+    expect(sortByString('world', 'hello', 'asc')).toBe(1);
+    expect(sortByString('Hello!', 'Hello')).toBe(1);
+    expect(sortByString('Linode 2', 'Linode 1')).toBe(1);
+    expect(sortByString('Linode 10', 'Linode 1')).toBe(1);
+  });
+
+  it('returns 0 when the values are equal', () => {
+    expect(sortByString('hello', 'hello')).toBe(0);
+    expect(sortByString('hello', 'hello', 'asc')).toBe(0);
+    expect(sortByString('hello', 'hello', 'desc')).toBe(0);
+  });
+
+  it('returns inverted values when sort order is descending', () => {
+    expect(sortByString('hello', 'world', 'desc')).toBe(1);
+    expect(sortByString('world', 'hello', 'desc')).toBe(-1);
+    expect(sortByString('hello', 'hello', 'desc')).toBe(0);
+  });
+});
+
+describe('sortByUTFDate', () => {
+  const dateA = '2022-01-01T00:00:00';
+  const dateB = '2022-06-01T12:15:30';
+  const dateC = '2022-06-02T00:00:00';
+  const dateD = '2023-01-01T00:00:00';
+
+  it('returns -1 when the first value comes first chronologically', () => {
+    expect(sortByUTFDate(dateA, dateB)).toBe(-1);
+    expect(sortByUTFDate(dateC, dateD, 'asc')).toBe(-1);
+    expect(sortByUTFDate(dateA, dateD)).toBe(-1);
+    expect(sortByUTFDate(dateB, dateC, 'asc')).toBe(-1);
+  });
+
+  it('returns 1 when the second value comes first chronologically', () => {
+    expect(sortByUTFDate(dateB, dateA)).toBe(1);
+    expect(sortByUTFDate(dateD, dateC, 'asc')).toBe(1);
+    expect(sortByUTFDate(dateD, dateA)).toBe(1);
+    expect(sortByUTFDate(dateC, dateB, 'asc')).toBe(1);
+  });
+
+  it('returns 0 when the values are equal', () => {
+    expect(sortByUTFDate(dateA, dateA)).toBe(0);
+    expect(sortByUTFDate(dateB, dateB, 'asc')).toBe(0);
+    expect(sortByUTFDate(dateC, dateC)).toBe(0);
+    expect(sortByUTFDate(dateD, dateD, 'desc')).toBe(0);
+  });
+
+  it('returns inverted values when the sort order is descending', () => {
+    expect(sortByUTFDate(dateA, dateB, 'desc')).toBe(1);
+    expect(sortByUTFDate(dateD, dateC, 'desc')).toBe(-1);
+    expect(sortByUTFDate(dateD, dateD, 'desc')).toBe(0);
+  });
```

Comment:
feedback: Can we follow similar pattern here? 
```suggestion
 it('Should sort sortByUTFDate default in asc order', () => {
    const actual = [
      '2022-01-01T00:00:00',
      '2023-01-01T00:00:00',
      '2022-06-02T00:00:00',
      '2022-06-01T12:15:30',
    ];
    const expected = [
      '2022-01-01T00:00:00',
      '2022-06-01T12:15:30',
      '2022-06-02T00:00:00',
      '2023-01-01T00:00:00',
    ];
    expect(actual.sort(sortByUTFDate)).toStrictEqual(expected);
  });
  it('Should sort alphabetically in desc order', () => {
    const actual = [
      '2022-01-01T00:00:00',
      '2022-06-01T12:15:30',
      '2022-06-02T00:00:00',
      '2023-01-01T00:00:00',
    ];
    const expected = [
      '2023-01-01T00:00:00',
      '2022-06-02T00:00:00',
      '2022-06-01T12:15:30',
      '2022-01-01T00:00:00',
    ];
    expect(actual.sort((a, b) => sortByUTFDate(a, b, 'desc'))).toStrictEqual(
      expected
    );
  });
```